terraform/gcp/modules/cloud_function_v2/function_v2.tf

```diff
@@ -0,0 +1,58 @@
+locals {
+  bucket_name = length(var.source_bucket) > 0 ? var.source_bucket : google_storage_bucket.default[0].name
+}
+
+resource "random_id" "default" {
+  byte_length = 8
+}
+
+resource "google_storage_bucket" "default" {
+  count = length(var.source_bucket) > 0 ? 0 : 1
+  name                        = "${random_id.default.hex}-gcf-source" # Every bucket name must be globally unique
+  location                    = var.location
+  uniform_bucket_level_access = true
+}
+
+data "archive_file" "default" {
+  type        = "zip"
+  output_path = "/tmp/function-source.zip"
+  source_dir  = "source"
```

Comment:
please set this via a variable, with default "source" , and change the README to reflect this. That way you can have multiple sources as needed, in a single run.